ark/phenotyping/som_utils.py

```diff
@@ -108,65 +191,32 @@ def create_pixel_matrix(img_xr, seg_labels, base_dir,
-        img_data_blur = img_xr.loc[fov, ...].values.astype(np.float32)
-
-        # for each marker, compute the Gaussian blur
-        for marker in range(len(img_xr.channels.values)):
-            img_data_blur[:, :, marker] = ndimage.gaussian_filter(img_data_blur[:, :, marker],
-                                                                  sigma=blur_factor)
-
-        # flatten each image
-        pixel_mat = img_data_blur.reshape(-1, len(img_xr.channels.values))
-
-        # convert into a dataframe
-        pixel_mat = pd.DataFrame(pixel_mat, columns=img_xr.channels.values)
-
-        # assign metadata about each entry
-        pixel_mat['fov'] = fov
-        pixel_mat['row_index'] = np.repeat(range(img_data_blur.shape[0]), img_data_blur.shape[1])
-        pixel_mat['column_index'] = np.tile(range(img_data_blur.shape[0]), img_data_blur.shape[1])
-
-        # assign segmentation label
-        seg_labels_flat = seg_labels.loc[fov, ...].values.flatten()
-        pixel_mat['segmentation_label'] = seg_labels_flat
-
-        # needed for selecting just channel columns
-        non_num_cols = ['fov', 'row_index', 'column_index', 'segmentation_label']
-        chan_cols = [col for col in pixel_mat.columns.values if col not in non_num_cols]
-
-        # remove any rows that sum to 0
-        pixel_mat = pixel_mat.loc[pixel_mat.loc[:, chan_cols].sum(axis=1) != 0, :]
-
-        # normalize each row by total marker counts to convert into frequencies
-        pixel_mat.loc[:, chan_cols] = pixel_mat.loc[:, chan_cols].div(
-            pixel_mat.loc[:, chan_cols].sum(axis=1), axis=0)
-
-        # subset the pixel matrix for training
-        pixel_mat_subset = pixel_mat.sample(frac=subset_proportion, random_state=seed)
-
-        # write complete dataset to feather, needed for cluster assignment
-        feather.write_dataframe(pixel_mat,
-                                os.path.join(base_dir,
-                                             pre_dir,
-                                             fov + ".feather"),
-                                compression='uncompressed')
-
-        # write subseted dataset to hdf5, needed for training
-        feather.write_dataframe(pixel_mat_subset,
-                                os.path.join(base_dir,
-                                             sub_dir,
-                                             fov + ".feather"),
-                                compression='uncompressed')
+        # load img_xr from MIBITiff or directory with the fov
+        if is_mibitiff:
+            img_xr = load_utils.load_imgs_from_mibitiff(
+                tiff_dir, mibitiff_files=[fov], channels=channels, dtype="int16"
+            )
+        else:
+            img_xr = load_utils.load_imgs_from_tree(
+                tiff_dir, fovs=[fov], channels=channels, dtype="int16"
+            )
+
+        # load segmentation labels in for fov
+        seg_labels = load_utils.load_imgs_from_dir(
```

Comment:
We can probably just load the raw image. I don't think we need it in an xarray for any reason, we just .values it immediately in the subsequent function